apps/tests/devapps/main.go

```diff
@@ -26,8 +26,20 @@ func main() {
 		panic("currently not implemented")
 		//acquireByAuthorizationCodeConfidential()
 	} else if exampleType == "5" {
+		// This sample does not use a serialized cache - it relies on in-memory cache by reusing the app object
+		// This works great for app tokens, because there is only 1 token per resource, per tenant and most
+		// developers only require 1-2 tokens.
+		acquireTokenClientSecret()
+
+		// this time the token comes from the cache!
 		acquireTokenClientSecret()
 	} else if exampleType == "6" {
+		// This samples	uses a serialized cache in a file. This is for demonstration purposes only of the caching interface.
```

Comment:
This comes as a surprise. Was it because the client_cert_sample contains `confidential.WithCache(cacheAccessor)`, while the client_secret_sample does not? Personally, I would prefer the two samples behave the same, i.e., both use in-memory cache or both use serialized cache. Both can contain same comments to elaborate when to use serialized cache.